lib/js/jsre.nim

```diff
@@ -53,11 +53,26 @@ func contains*(pattern: cstring; self: RegExp): bool =
 
 
 runnableExamples:
-  let jsregex: RegExp = newRegExp(r"\s+", r"i")
-  jsregex.compile(r"\w+", r"i")
-  assert "nim javascript".contains jsregex
-  assert jsregex.exec(r"nim javascript") == @["nim".cstring]
-  assert jsregex.toCstring() == r"/\w+/i"
-  jsregex.compile(r"[0-9]", r"i")
-  assert "0123456789abcd".contains jsregex
-  assert $jsregex == "/[0-9]/i"
+  block:
+    let jsregex: RegExp = newRegExp(r"\s+", r"i")
+    jsregex.compile(r"\w+", r"i")
+    assert "nim javascript".contains jsregex
+    assert jsregex.exec(r"nim javascript") == @["nim".cstring]
+    assert jsregex.toCstring() == r"/\w+/i"
+    jsregex.compile(r"[0-9]", r"i")
+    assert "0123456789abcd".contains jsregex
+    assert $jsregex == "/[0-9]/i"
+
+  ## "Global" flag may or may not cause "alternating" behaviour.
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp#the_difference_between_the_sticky_flag_and_the_global_flag
+  block:
+    let regex = newRegExp("a", "g")
+    assert "a".contains regex
+    assert not("a".contains regex)
+    assert "a".contains regex
+    assert not("a".contains regex)
```

Comment:
This looks super strange, the result changes with each call?

Reply:
JavaScript Shenanigans :shrug:

Reply:
is not your bug, is not my bug, is just how things are, Nim is doing the right thing, even if its a little weird  :)

Reply:
Don't document design bugs inherited from JS's libraries.

Reply:
Ok, closing.

Reply:
@Araq there's a reason it works this way though, although it looks bizarre in this example, in more realistic scenarios it allows you to iterate over matches starting with last position

And this is `std/jsre`, not `std/re` it makes sense to document it. TIL about this behavior thanks to @juancarlospaco 's PR. IMO we should re-open and fix whatever needs to be fixed in this PR